### PR TITLE
get background genes for pathway anal

### PR DESCRIPTION
**Pull request by alexvpickering**
# Background
#### Link to issue 

https://biomage.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=BIOMAGE&modal=detail&selectedIssue=BIOMAGE-1255

#### Anything else the reviewers should know about the changes here

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock (with latest production data downloaded with biomage experiment pull)
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

File: r/R/get_background_genes.R [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    ntot <- Matrix::rowSums(data[['RNA']]@counts)
+    keep <- ntot > min.total.count
+
+    genes <- data@misc$gene_annotations$original_name[keep]
+
+    return(list(genes = genes))
+}
+
+# adds 'custom' slot to identitify background and base cells
+add_comparison_group <- function(req, data) {
```
Comment (cosa65) [inline r/R/get_background_genes.R]: This function seems to follow a different naming convention than the rest of the functions, example:

`getBackgroundExpressedGenes` (in camelcase) vs `add_comparison_group` (in underscore)

File: r/work.R [hunk 2]
```diff
@@ -155,6 +155,13 @@ create_app <- function(last_modified, data, fpath) {
             res$set_body(result)
         }
     )
+    app$add_post(
+        path = "/v0/getBackgroundExpressedGenes",
+        FUN = function(req, res) {
```
Comment (cosa65) [inline r/work.R:160]: Out of curiosity, does the `FUN` variable do anything special?

File: r/R/get_background_genes.R [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+#' @param data SeuratObject
+#'
+#' @return list with gene symbols that are expressed in cells being compared.
+#' @export
+#'
+getBackgroundExpressedGenes <- function(req, data) {
+
+    # minimum total count for gene to be considered expressed
+    # from edgeR::filterByExpr
+    min.total.count = 15
```
Comment (cosa65) [inline r/R/get_background_genes.R:13]: Just a suggestion so feel free to ignore: is this a special constant value that we might use in other places? If it is and we have other constants of this type we could move them all to a `constants` file that we use across the project.

**Review by ogibson — APPROVED** [review]

File: r/work.R (hunk 2 above)
Comment (alexvpickering) [reply r/work.R:160]: It's the function that get called from the post request

File: r/R/get_background_genes.R (hunk 3 above)
Comment (alexvpickering) [reply r/R/get_background_genes.R:13]: Agreed. There are a few of these lingering constants so I'll push this to another task.

**Pull request merged**